### PR TITLE
AB#3889 - frontend mock create new confirmation code

### PR DESCRIPTION
**Pull request by leiye29**
### Description

frontend mock api to create a new confirmation code for subscription email validation

### Related Azure Boards Work Items

[AB#3889](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3889)
[AB#3886](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3886)

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/services/subscription-service.server.ts [hunk 1]
```diff
@@ -309,39 +312,62 @@ function createSubscriptionService() {
...
+        statusText: confirmationCodeResponse.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to request data. Status: ${confirmationCodeResponse.status}, Status Text: ${confirmationCodeResponse.statusText}`);
     }
 
-    return response;
+    return confirmationCodeResponse;
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/services/subscription-service.server.ts]: removed

File: frontend/app/services/subscription-service.server.ts [hunk 2]
```diff
@@ -309,39 +312,60 @@ function createSubscriptionService() {
...
+
+    const confirmationCodesUrl = userParsed._links.confirmationCodes.href;
+    const confirmationCodeResponse = await fetch(confirmationCodesUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.codes.request.posts', confirmationCodeResponse.status);
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/services/subscription-service.server.ts]: fixed

File: frontend/app/services/subscription-service.server.ts [hunk 3]
```diff
@@ -180,7 +183,7 @@ function createSubscriptionService() {
       },
     });
 
-    instrumentationService.countHttpStatus('http.client.cdcp-api.alert-subscription.delete', subscriptionsResponse.status);
+    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.deletes', subscriptionsResponse.status);
```
Comment (nicholas-ly) [inline frontend/app/services/subscription-service.server.ts:186]:
```suggestion
    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.gets', subscriptionsResponse.status);
```

Comment (leiye29) [reply frontend/app/services/subscription-service.server.ts]: fixed

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**